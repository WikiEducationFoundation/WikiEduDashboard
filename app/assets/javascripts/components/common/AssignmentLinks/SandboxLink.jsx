import React from 'react';
import PropTypes from 'prop-types';

export const SandboxLink = ({ assignment, editMode }) => {
  let url = assignment.sandboxUrl || assignment.sandbox_url;
  if (Features.wikiEd && editMode) {
    url += '?veaction=edit&preload=Template:Dashboard.wikiedu.org_draft_template';
  }
  return (
    <a href={url} target="_blank">
      {I18n.t('assignments.sandbox_draft_link')}
    </a>
  );
};

SandboxLink.propTypes = {
  // props
  assignment: PropTypes.object.isRequired,
};

export default SandboxLink;
