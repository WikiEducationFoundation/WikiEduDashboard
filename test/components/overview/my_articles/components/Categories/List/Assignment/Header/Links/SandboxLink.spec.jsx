import React from 'react';
import { shallow } from 'enzyme';
import '../../../../../../../../../testHelper';

import SandboxLink from '~/app/assets/javascripts/components/common/AssignmentLinks/SandboxLink';

describe('SandboxLink', () => {
  Features.wikiEd = true;

  it('should show the link', () => {
    const component = shallow(<SandboxLink assignment={{ sandboxUrl: 'url' }} />);
    expect(component).toMatchSnapshot();
    expect(component.find('a').props().href).toContain('url');
  });

  it('should include a template in the link if the assignment is a new article and editMode is true', () => {
    const component = shallow(<SandboxLink assignment={{ status: 'new_article' }} editMode={true} />);
    expect(component).toMatchSnapshot();
    expect(component.find('a').props().href).toContain('Template');
  });
});
