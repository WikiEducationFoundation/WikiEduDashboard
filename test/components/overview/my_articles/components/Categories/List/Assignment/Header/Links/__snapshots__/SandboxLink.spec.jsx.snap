// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SandboxLink should include a template in the link if the assignment is a new article and editMode is true 1`] = `ShallowWrapper {}`;

exports[`SandboxLink should show the link 1`] = `ShallowWrapper {}`;
